main.cc

```diff
@@ -919,6 +920,13 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
                 }).get();
             }
 
+            static sharded<service::raft_address_map<>> raft_address_map;
```

Comment:
Is it necessarily that it lives in main? AFAICS it (used to) live as `raft_group_registry::_srv_address_mappings`, and group registry is already peering service. Can _it_ be taught to replicate its address map?

Reply:
OK, since it's going to be shared between several users, then it makes sense.

Reply:
`raft_address_map` maps Raft server IDs to their last known IP addresses (currently we assume static mapping, but @kostja is changing that - see https://github.com/scylladb/scylladb/pull/11635)
`topology` maps IP addresses to their dc/rack information.

Seems they have different purposes, not sure if merging them at this point would make sense.

However, suppose in the future we update everything, including `topology`, to not operate in terms of IP addresses but use Raft server IDs - which could become general "node unique immutable ID" (we didn't have such IDs previously yet - Host IDs came close but it can be stolen during node replace operation). Then `topology` would map not IP addrs, but server IDs to dc/rack information. Then we could merge the two: `topology` would map dc/rack to both the last known IP addr and dc/rack (and maybe other information about the node).

Reply:
> topology maps IP addresses to their dc/rack information.

... and back

> raft_address_map maps Raft server IDs to their last known IP addresses

Does "raft server ID" have something with `host_id`?

Reply:
> Does "raft server ID" have something with host_id?

They are both UUIDs, but :
- they are generated independently
- there is the main difference (which is why we couldn't reuse Host IDs to identify Raft servers): Host ID can be assigned to different node during replace operation, Raft server ID can't.

Reply:
@kbr- for what it's worth, I'd keep the map in the registry. In my value set, good modularity is when there are not too many medium-size modules, not tons of tiny modules with potentially hairy dependencies. Even if these dependencies are strictly necessary, explicit and expressed in main(). Just take a look at the history of de-globalization in Scylla: it literally exploded main with lots and lots of new variables in main(). 

But I definitely do not want to insist on changing the pull request for the sake of it. My view at architecture is evolutionary. Take a look at a village or a city. Houses and streets are built, torn down, demolished. Over time the face of a city emerges. This is very similar to what is happening with modularity in Scylla now. Perhaps eventually we will merge the snitch into raft_address_map as @xemul suggests, in this case keeping the map global would be a wise choice.